### PR TITLE
Issue #3210: fix compact_no_array in AnnotationUseStyleCheck

### PR DESCRIPTION
**Pull request by strkkk**
#3210

Reg. report: https://strkkk.github.io/3210/

### CODE REVIEW

**Comment by rnveach** [comment]
pitest is failing.

**Comment by strkkk** [comment]
> pitest is failing.

Fixed

**Comment by strkkk** [comment]
Reg report added in first message

**Comment by romani** [comment]
@pbludov , please do first review.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java [hunk 1]
```diff
@@ -428,28 +426,24 @@ private void checkCompactNoArrayStyle(final DetailAST annotation) {
...
-        final int valuePairCount =
-            annotation.getChildCount(TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR);
-
         //in compact style with one value
         if (arrayInit != null
             && arrayInit.getChildCount(TokenTypes.EXPR) == 1) {
             log(annotation.getLineNo(), MSG_KEY_ANNOTATION_INCORRECT_STYLE,
                 ElementStyle.COMPACT_NO_ARRAY);
         }
         //in expanded style with one value and the correct element name
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java]: This comment should be changed too.

**Comment by pbludov** [comment]
> Reg. report: https://strkkk.github.io/3210/

Regression looks good.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java]: done

**Comment by strkkk** [comment]
CI is ok, travis failed with some infra issue

**Comment by romani** [comment]
@strkkk , 
please help to address xdoc improvement , requested at https://github.com/checkstyle/checkstyle/issues/3210#issuecomment-326997893
 
>Please update section - https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/property_types.xml#L512 to avoid any future misinterpretations of how works.

**Comment by strkkk** [comment]
@romani I made changes in separate commit to make changes clearer. Please take a look

**Comment by romani** [comment]
please rebase to fix Travis

**Comment by strkkk** [comment]
> please rebase to fix Travis.

done, ci is green

**Pull request merged**